central/group/datastore/datastore_impl.go

```diff
@@ -95,7 +134,33 @@ func (ds *dataStoreImpl) Update(ctx context.Context, group *storage.Group) error
 		return errox.InvalidArgs.CausedBy(err)
 	}
 
-	return ds.storage.Update(group)
+	// Lock to ensure synchronization between the get and upsert
+	ds.lock.Lock()
+	defer ds.lock.Unlock()
+
+	// TODO(ROX-11592): Once the deprecation of retrieving groups by their properties is fully deprecated, this condition
+	// can be removed and groups shall only be retrievable via their id.
+	if group.GetProps().GetId() == "" {
+		id, err := ds.findPropsIDByProps(ctx, group.GetProps())
+		if err != nil {
+			return err
+		}
+		// Use the id of the retrieved group to update
+		group.GetProps().Id = id
+	}
+
+	defaultGroup, err := ds.getDefaultGroupForProps(ctx, group.GetProps())
+	if err != nil {
+		return err
+	}
+
+	// Only disallow update of a default group if it does not update the existing default group, if there is any.
+	if defaultGroup != nil && defaultGroup.GetProps().GetId() != group.GetProps().Id {
+		return errox.AlreadyExists.Newf("a default group already exists for auth provider %q",
```

Comment:
The error does not capture the intent correctly.

Reply:
I copied this directly from existing code. In this case, user already has a default group and they are trying update another group to a default group. What would you suggest?

Maybe, "cannot update groups <id> to default group as a default group already exists for auth provider %q"?